### ISSUE TITLE
Parallel execution doesn't preserve method dependencies for classes in hierarchy

### ISSUE THREAD

**Issue by jimirocks** [testng-team/testng#992]
Let's take code similar to this:

```
public abstract class AbstractTest {

    @Test
    public void testCommon() throws Exception {
        .....
    }
}

public class SpecificTest1 extends AbstractTest {

    @Test(dependsOnMethods = "testCommon")
    public void testSpecific1() throws Exception {
        // should wait for the same instance testCommon method
    }
}

public class SpecificTest2 extends AbstractTest {

    @Test(dependsOnMethods = "testCommon")
    public void testSpecific2() throws Exception {
        // should wait for the same instance testCommon method
    }
}
```

and execute it in parallel (no matter what parallel setting). It can happen that `testSpecific2` will run while `SpecificTest2.testCommon` - which is _WRONG_, making parallel execution with test dependencies almost useless

**Comment by juherr** [comment]
Closing as resolved in the current implementation.

This issue was caused by method dependencies being resolved across all test instances. `MethodHelper` now resolves `dependsOnMethods` against methods belonging to the same test instance first, which prevents an inherited dependency from being satisfied by another subclass instance during parallel execution.

I could not find a PR explicitly linked to #992, and I did not rerun the original 2016 reproducer. If the same behavior can still be reproduced with a current TestNG release, please open a new issue with an up-to-date minimal reproducer.
